packages/clay-core/src/tree-view/TreeViewItem.tsx

```diff
@@ -86,6 +91,23 @@ export const TreeViewItem = React.forwardRef<HTMLDivElement, TreeViewItemProps>(
 								}
 							}
 						}}
+						onKeyDown={(event) => {
+							const {key} = event;
+							if (group) {
+								if (key === 'ArrowLeft') {
+									close(item.key);
+								}
+								if (key === 'ArrowRight') {
+									open(item.key);
+								}
+							}
+							if (key === 'Backspace' || key === 'Delete') {
+								remove(item.indexes);
+							}
+							if ((key === 'R' || key === 'F2') && onRenameItem) {
+								rename(item.indexes, onRenameItem(item.name));
+							}
```

Comment:
We could try using a `switch case` here to avoid the number of conditionals, for example:

```js
switch (key) {
	case 'ArrowLeft': {
		if (group) {
			close(item.key);
		}
		break;
	}
	case 'ArrowRight': {
		if (group) {
			open(item.key);
		}
		break;
	}
	case 'Delete':
	case 'Backspace':
		remove(item.indexes);
		break;
	case 'F2':
	case 'R':
		rename(item.indexes);
		break;
	default:
		break;
}
```

Reply:
We could, but I see the same duplicated `if` condition which makes me say it's not "better" nor is it more readable - change my mind :smile: .